### PR TITLE
fix bug for pandas auto categorical_feature

### PR DESCRIPTION
**Pull request by wxchan**
#215

### CODE REVIEW

**Comment by wxchan** [comment]
still one bug, pandas feature names can be int, how to distinguish between feature names (if it's int) and feature index when set categorical features?

**Comment by henry0312** [comment]
@wxchan How about using `pandas.get_dummies` ?
cf. http://pandas.pydata.org/pandas-docs/stable/generated/pandas.get_dummies.html

**Comment by wxchan** [comment]
@henry0312 isn't that one-hot encoding? it's not how to deal with categorical feature in lightgbm

**Comment by henry0312** [comment]
hum, I'll check carefully.

**Comment by wxchan** [comment]
@guolinke this pr is finished too, I am not sure about all the logic, it may need some review.

File: python-package/lightgbm/basic.py [hunk 1]
```diff
@@ -1196,6 +1201,12 @@ def __init__(self, params=None, train_set=None, model_file=None, silent=False):
                 self.handle,
                 ctypes.byref(out_num_class)))
             self.__num_class = out_num_class.value
+            with open(model_file, 'r') as f:
+                last_line = f.readlines()[-1]
+                if last_line.startswith('pandas_categorical:'):
+                    self.pandas_categorical = eval(last_line[len('pandas_categorical:'):])
+                else:
+                    self.pandas_categorical = None
```
Comment (guolinke) [inline python-package/lightgbm/basic.py]: I think maybe it need to this part on constructor of _inner_predictor as well

Comment (wxchan) [inline python-package/lightgbm/basic.py]: seem no places use model_file to init _InnerPredictor?

Comment (wxchan) [inline python-package/lightgbm/basic.py]: it is in engine.py, will add it later

Comment (guolinke) [inline python-package/lightgbm/basic.py]: ok, it seems be.. But I we better keep them by same logic.

**Pull request merged**